This notebook is for robustness check for linear regression. In this notebook, we replace the time variable (post) to yearly variables. We will still pay attention to the interaction terms between yearly variables and treatment variable (round 1). If the sign of the interaction term is negative after 2008, then SSfS is efficiently decreasing the rate of accident.

Conclusion:
- Similar result compare to our main regression model
- Spill over effect exist

In [1]:
import pandas as pd
import statsmodels as sm
import statsmodels.formula.api as smf
import statsmodels.api as sma

In [2]:
ct = pd.read_csv("ct_df.csv")
ct.head()

Unnamed: 0  SENIOR  S_PED  S_NOT_PED  S_PED_DEAD  S_SURVIVED  S_DEAD  \
0       62200     0.0    0.0        0.0         0.0         0.0     0.0   
1       19300     1.0    0.0        1.0         0.0         0.0     1.0   
2       24100     3.0    1.0        2.0         1.0         0.0     3.0   
3       82000     0.0    0.0        0.0         0.0         0.0     0.0   
4       16800     2.0    2.0        0.0         2.0         0.0     2.0   

   S_DRINKING+  S_DRINKING-  S_MALE  ...   AFTERNOON  NIGHT  MIDNIGHT  \
0          0.0          0.0     0.0  ...         0.0    0.0       0.0   
1          1.0          0.0     1.0  ...         1.0    0.0       0.0   
2          0.0          1.0     3.0  ...         1.0    0.0       0.0   
3          0.0          0.0     0.0  ...         0.0    1.0       0.0   
4          0.0          2.0     1.0  ...         0.0    0.0       0.0   

   MONTH_1  MONTH_2  MONTH_3  rounds_1.0  rounds_2.0  rounds_3.0  YEAR  
0      0.0      1.0      0.0         0.0         0.0         0.0  2001  
1      0.0      1.0      0.0         0.0         0.0         1.0  2001  
2      0.0      1.0      0.0         1.0         0.0         0.0  2001  
3      0.0      1.0      0.0         0.0         1.0         0.0  2001  
4      0.0      0.0      1.0         0.0         0.0         0.0  2001  

[5 rows x 62 columns]

In [3]:
ct.rename(columns={"Unnamed: 0":"ct"},inplace=True)

In [4]:
ct.set_index("ct",inplace=True)

In [5]:
ct.S_PED_DEAD.sum()

916.0

In [6]:
ct.rename(columns={"rounds_1.0":"rounds_1","rounds_2.0":"rounds_2","rounds_3.0":"rounds_3"},inplace=True)

In [7]:
ct.columns

Index(['SENIOR', 'S_PED', 'S_NOT_PED', 'S_PED_DEAD', 'S_SURVIVED', 'S_DEAD',
       'S_DRINKING+', 'S_DRINKING-', 'S_MALE', 'S_FEMALE', 'S_DRUG+',
       'S_DRUG-', 'S_DEAD_AFTER', 'S_DEAD_SCENE', 'S_DEAD_ROUTE', 'YOUNG',
       'Y_PED', 'Y_NOT_PED', 'Y_SURVIVED', 'Y_DEAD', 'Y_PED_DEAD',
       'Y_DRINKING+', 'Y_DRINKING-', 'Y_MALE', 'Y_FEMALE', 'Y_DRUG+',
       'Y_DRUG-', 'Y_DEAD_AFTER', 'Y_DEAD_SCENE', 'Y_DEAD_ROUTE', 'FATALS',
       'WEATHER_GOOD', 'WEATHER_RAIN', 'WEATHER_SLEET', 'WEATHER_SNOW',
       'WEATHER_FOG', 'WEATHER_CLOUDY', 'LGT_COND_DAYLIGHT',
       'LGT_COND_DARK_NOT_LIGHTED', 'LGT_COND_DARK_LIGHTED', 'LGT_COND_DAWN',
       'LGT_COND_DUSK', 'LGT_COND_DARK_UNKNOWN_LIGHTING', 'WEEKDAY', 'WEEKEND',
       'YEAR_Q1', 'YEAR_Q2', 'YEAR_Q3', 'MONTH_Q4', 'MORNING', 'NOON',
       'AFTERNOON', 'NIGHT', 'MIDNIGHT', 'MONTH_1', 'MONTH_2', 'MONTH_3',
       'rounds_1', 'rounds_2', 'rounds_3', 'YEAR'],
      dtype='object')

- YEAR_Q1: January, Febuary, March
- YEAR_Q2: April, May, June
- YEAR_Q3: July, August, September
- YEAR_Q4: October, November, December
- MORNING: 6-11
- NOON: 12-14
- AFTERNOON: 15-19
- NIGHT: 20-23
- MIDNIGHT: 0-5
- MONTH_1: 1-10
- MONTH_2: 11-20
- MONTH_3: 21-31

# Y = Number of senior-involved accidents 

In [8]:
mod = smf.ols(formula="SENIOR ~ C(YEAR)+C(YEAR)*rounds_1", data = ct).fit()
print(mod.summary())

                            OLS Regression Results                            
Dep. Variable:                 SENIOR   R-squared:                       0.076
Model:                            OLS   Adj. R-squared:                  0.074
Method:                 Least Squares   F-statistic:                     37.38
Date:                Thu, 11 Jul 2019   Prob (F-statistic):          4.21e-228
Time:                        13:29:10   Log-Likelihood:                -6823.6
No. Observations:               15028   AIC:                         1.372e+04
Df Residuals:                   14994   BIC:                         1.397e+04
Df Model:                          33                                         
Covariance Type:            nonrobust                                         
                               coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------
Intercept               

- R-squared: 0.076
- Positive interaction terms:2005,2015,**2017** 
- Not significant interaction terms: 2004, 2005, 2006, 2008, 2010, 2015

In [9]:
ct.columns

Index(['SENIOR', 'S_PED', 'S_NOT_PED', 'S_PED_DEAD', 'S_SURVIVED', 'S_DEAD',
       'S_DRINKING+', 'S_DRINKING-', 'S_MALE', 'S_FEMALE', 'S_DRUG+',
       'S_DRUG-', 'S_DEAD_AFTER', 'S_DEAD_SCENE', 'S_DEAD_ROUTE', 'YOUNG',
       'Y_PED', 'Y_NOT_PED', 'Y_SURVIVED', 'Y_DEAD', 'Y_PED_DEAD',
       'Y_DRINKING+', 'Y_DRINKING-', 'Y_MALE', 'Y_FEMALE', 'Y_DRUG+',
       'Y_DRUG-', 'Y_DEAD_AFTER', 'Y_DEAD_SCENE', 'Y_DEAD_ROUTE', 'FATALS',
       'WEATHER_GOOD', 'WEATHER_RAIN', 'WEATHER_SLEET', 'WEATHER_SNOW',
       'WEATHER_FOG', 'WEATHER_CLOUDY', 'LGT_COND_DAYLIGHT',
       'LGT_COND_DARK_NOT_LIGHTED', 'LGT_COND_DARK_LIGHTED', 'LGT_COND_DAWN',
       'LGT_COND_DUSK', 'LGT_COND_DARK_UNKNOWN_LIGHTING', 'WEEKDAY', 'WEEKEND',
       'YEAR_Q1', 'YEAR_Q2', 'YEAR_Q3', 'MONTH_Q4', 'MORNING', 'NOON',
       'AFTERNOON', 'NIGHT', 'MIDNIGHT', 'MONTH_1', 'MONTH_2', 'MONTH_3',
       'rounds_1', 'rounds_2', 'rounds_3', 'YEAR'],
      dtype='object')

In [10]:
ct.rename(columns={"S_DRINKING+":"S_DRINKING","S_DRUG+":"S_DRUG"},inplace=True)

In [11]:
mod1 = smf.ols(formula="SENIOR ~ C(YEAR)+C(YEAR)*rounds_1+S_DRINKING+S_DRUG+WEATHER_GOOD+\
        WEATHER_RAIN+WEATHER_SLEET+WEATHER_SNOW+WEATHER_FOG+WEATHER_CLOUDY+LGT_COND_DAYLIGHT\
       +LGT_COND_DARK_NOT_LIGHTED+LGT_COND_DARK_LIGHTED+LGT_COND_DAWN+\
       LGT_COND_DUSK+LGT_COND_DARK_UNKNOWN_LIGHTING+WEEKDAY+\
       YEAR_Q1+YEAR_Q2+YEAR_Q3+MORNING+NOON+\
       AFTERNOON+NIGHT", data = ct).fit()
print(mod1.summary())

                            OLS Regression Results                            
Dep. Variable:                 SENIOR   R-squared:                       0.483
Model:                            OLS   Adj. R-squared:                  0.481
Method:                 Least Squares   F-statistic:                     254.6
Date:                Thu, 11 Jul 2019   Prob (F-statistic):               0.00
Time:                        13:29:11   Log-Likelihood:                -2456.5
No. Observations:               15028   AIC:                             5025.
Df Residuals:                   14972   BIC:                             5452.
Df Model:                          55                                         
Covariance Type:            nonrobust                                         
                                     coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------------
Intercept   

- R-squared: 0.483
- Positive interaction terms: 2005,2015, **2017**
- Not significant interaction terms: 2005, 2006, 2008, 2010, 2015
- Not significant covariance: rain, night

In [12]:
ct.columns

Index(['SENIOR', 'S_PED', 'S_NOT_PED', 'S_PED_DEAD', 'S_SURVIVED', 'S_DEAD',
       'S_DRINKING', 'S_DRINKING-', 'S_MALE', 'S_FEMALE', 'S_DRUG', 'S_DRUG-',
       'S_DEAD_AFTER', 'S_DEAD_SCENE', 'S_DEAD_ROUTE', 'YOUNG', 'Y_PED',
       'Y_NOT_PED', 'Y_SURVIVED', 'Y_DEAD', 'Y_PED_DEAD', 'Y_DRINKING+',
       'Y_DRINKING-', 'Y_MALE', 'Y_FEMALE', 'Y_DRUG+', 'Y_DRUG-',
       'Y_DEAD_AFTER', 'Y_DEAD_SCENE', 'Y_DEAD_ROUTE', 'FATALS',
       'WEATHER_GOOD', 'WEATHER_RAIN', 'WEATHER_SLEET', 'WEATHER_SNOW',
       'WEATHER_FOG', 'WEATHER_CLOUDY', 'LGT_COND_DAYLIGHT',
       'LGT_COND_DARK_NOT_LIGHTED', 'LGT_COND_DARK_LIGHTED', 'LGT_COND_DAWN',
       'LGT_COND_DUSK', 'LGT_COND_DARK_UNKNOWN_LIGHTING', 'WEEKDAY', 'WEEKEND',
       'YEAR_Q1', 'YEAR_Q2', 'YEAR_Q3', 'MONTH_Q4', 'MORNING', 'NOON',
       'AFTERNOON', 'NIGHT', 'MIDNIGHT', 'MONTH_1', 'MONTH_2', 'MONTH_3',
       'rounds_1', 'rounds_2', 'rounds_3', 'YEAR'],
      dtype='object')

In [13]:
mod2 = smf.ols(formula="SENIOR ~ C(YEAR)+C(YEAR)*rounds_1+S_DRINKING+S_DRUG+WEATHER_GOOD+\
        WEATHER_RAIN+WEATHER_SLEET+WEATHER_SNOW+WEATHER_FOG+WEATHER_CLOUDY+LGT_COND_DAYLIGHT\
       +LGT_COND_DARK_NOT_LIGHTED+LGT_COND_DARK_LIGHTED+LGT_COND_DAWN+\
       LGT_COND_DUSK+LGT_COND_DARK_UNKNOWN_LIGHTING+WEEKDAY+\
       YEAR_Q1+YEAR_Q2+YEAR_Q3+NOON+\
       AFTERNOON+NIGHT+rounds_2+rounds_3", data = ct).fit()
print(mod2.summary())

                            OLS Regression Results                            
Dep. Variable:                 SENIOR   R-squared:                       0.485
Model:                            OLS   Adj. R-squared:                  0.483
Method:                 Least Squares   F-statistic:                     252.2
Date:                Thu, 11 Jul 2019   Prob (F-statistic):               0.00
Time:                        13:29:12   Log-Likelihood:                -2425.5
No. Observations:               15028   AIC:                             4965.
Df Residuals:                   14971   BIC:                             5399.
Df Model:                          56                                         
Covariance Type:            nonrobust                                         
                                     coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------------
Intercept   

- R-squared: 0.485
- Positive interaction terms: 2005,**2017**
- Not significant interaction terms: 2005, 2006, 2008, 2010, 2015
- Not significant covariance: rain, afternoon

In [14]:
mod3 = smf.ols(formula="SENIOR ~ C(YEAR)+C(YEAR)*rounds_1+S_DRINKING+S_DRUG+WEATHER_GOOD+\
        WEATHER_RAIN+WEATHER_SLEET+WEATHER_SNOW+WEATHER_FOG+WEATHER_CLOUDY+LGT_COND_DAYLIGHT\
       +LGT_COND_DARK_NOT_LIGHTED+LGT_COND_DARK_LIGHTED+LGT_COND_DAWN+\
       LGT_COND_DUSK+LGT_COND_DARK_UNKNOWN_LIGHTING+WEEKDAY+\
       YEAR_Q1+YEAR_Q2+YEAR_Q3+MORNING+NOON+\
       AFTERNOON+NIGHT+rounds_2+rounds_3+S_MALE", data = ct).fit()
print(mod3.summary())

                            OLS Regression Results                            
Dep. Variable:                 SENIOR   R-squared:                       0.690
Model:                            OLS   Adj. R-squared:                  0.688
Method:                 Least Squares   F-statistic:                     573.4
Date:                Thu, 11 Jul 2019   Prob (F-statistic):               0.00
Time:                        13:29:13   Log-Likelihood:                 1373.2
No. Observations:               15028   AIC:                            -2628.
Df Residuals:                   14969   BIC:                            -2179.
Df Model:                          58                                         
Covariance Type:            nonrobust                                         
                                     coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------------
Intercept   

- R-squared: 0.690
- Positive interaction terms: 2015
- Not significant interaction terms: 2004, 2005, 2010, 2015
- Not significant covariance: drinking, drug, all weather related factors, dark and unknown lighting, weekday, morning, afternoon, round2

# Y = Number of dead senior pedestrain

In [15]:
mod4 = smf.ols(formula="S_PED_DEAD ~ C(YEAR)+C(YEAR)*rounds_1", data = ct).fit()
print(mod4.summary())

                            OLS Regression Results                            
Dep. Variable:             S_PED_DEAD   R-squared:                       0.077
Model:                            OLS   Adj. R-squared:                  0.075
Method:                 Least Squares   F-statistic:                     38.14
Date:                Thu, 11 Jul 2019   Prob (F-statistic):          4.79e-233
Time:                        13:29:13   Log-Likelihood:                -34.696
No. Observations:               15028   AIC:                             137.4
Df Residuals:                   14994   BIC:                             396.4
Df Model:                          33                                         
Covariance Type:            nonrobust                                         
                               coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------
Intercept               

- R-squared: 0.077
- Positive interaction terms: 2005, 2006, **2017**
- Not significant interaction terms: 2004, 2005, 2006, 2007, 2010, 2013, 2015

In [16]:
mod5 = smf.ols(formula="S_PED_DEAD ~ C(YEAR)+C(YEAR)*rounds_1+S_DRINKING+S_DRUG+WEATHER_GOOD+\
        WEATHER_RAIN+WEATHER_SLEET+WEATHER_SNOW+WEATHER_FOG+WEATHER_CLOUDY+LGT_COND_DAYLIGHT\
       +LGT_COND_DARK_NOT_LIGHTED+LGT_COND_DARK_LIGHTED+LGT_COND_DAWN+\
       LGT_COND_DUSK+LGT_COND_DARK_UNKNOWN_LIGHTING+WEEKDAY+\
       YEAR_Q1+YEAR_Q2+YEAR_Q3+MORNING+NOON+\
       AFTERNOON+NIGHT", data = ct).fit()
print(mod5.summary())

                            OLS Regression Results                            
Dep. Variable:             S_PED_DEAD   R-squared:                       0.444
Model:                            OLS   Adj. R-squared:                  0.442
Method:                 Least Squares   F-statistic:                     217.5
Date:                Thu, 11 Jul 2019   Prob (F-statistic):               0.00
Time:                        13:29:14   Log-Likelihood:                 3771.9
No. Observations:               15028   AIC:                            -7432.
Df Residuals:                   14972   BIC:                            -7005.
Df Model:                          55                                         
Covariance Type:            nonrobust                                         
                                     coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------------
Intercept   

- R-squared: 0.444
- Positive interaction terms: 2006,**2017**
- Not significant interaction terms: 2004, 2005, 2006, 2015
- Not significant covariance: good weather, rain, sleet, snow, cloudy, year_Q1, night

In [17]:
mod6 = smf.ols(formula="S_PED_DEAD ~ C(YEAR)+C(YEAR)*rounds_1+S_DRINKING+S_DRUG+WEATHER_GOOD+\
        WEATHER_RAIN+WEATHER_SLEET+WEATHER_SNOW+WEATHER_FOG+WEATHER_CLOUDY+LGT_COND_DAYLIGHT\
       +LGT_COND_DARK_NOT_LIGHTED+LGT_COND_DARK_LIGHTED+LGT_COND_DAWN+\
       LGT_COND_DUSK+LGT_COND_DARK_UNKNOWN_LIGHTING+WEEKDAY+\
       YEAR_Q1+YEAR_Q2+YEAR_Q3+MORNING+NOON+\
       AFTERNOON+NIGHT+rounds_2+rounds_3", data = ct).fit()
print(mod6.summary())

                            OLS Regression Results                            
Dep. Variable:             S_PED_DEAD   R-squared:                       0.453
Model:                            OLS   Adj. R-squared:                  0.451
Method:                 Least Squares   F-statistic:                     217.5
Date:                Thu, 11 Jul 2019   Prob (F-statistic):               0.00
Time:                        13:29:15   Log-Likelihood:                 3893.1
No. Observations:               15028   AIC:                            -7670.
Df Residuals:                   14970   BIC:                            -7228.
Df Model:                          57                                         
Covariance Type:            nonrobust                                         
                                     coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------------
Intercept   

- R-squared: 0.453
- Positive interaction terms: 2006,**2017**
- Not significant interaction terms: 2004, 2005, 2006, 2015
- Not significant covariance: good weather, rain, sleet, snow, cloudy, year_Q1, night

In [18]:
mod7 = smf.ols(formula="S_PED_DEAD ~ C(YEAR)+C(YEAR)*rounds_1+S_DRINKING+S_DRUG+WEATHER_GOOD+\
        WEATHER_RAIN+WEATHER_SLEET+WEATHER_SNOW+WEATHER_FOG+WEATHER_CLOUDY+LGT_COND_DAYLIGHT\
       +LGT_COND_DARK_NOT_LIGHTED+LGT_COND_DARK_LIGHTED+LGT_COND_DAWN+\
       LGT_COND_DUSK+LGT_COND_DARK_UNKNOWN_LIGHTING+WEEKDAY+\
       YEAR_Q1+YEAR_Q2+YEAR_Q3+MORNING+NOON+\
       AFTERNOON+NIGHT+rounds_2+rounds_3+S_MALE", data = ct).fit()
print(mod7.summary())

                            OLS Regression Results                            
Dep. Variable:             S_PED_DEAD   R-squared:                       0.576
Model:                            OLS   Adj. R-squared:                  0.574
Method:                 Least Squares   F-statistic:                     350.7
Date:                Thu, 11 Jul 2019   Prob (F-statistic):               0.00
Time:                        13:29:15   Log-Likelihood:                 5807.6
No. Observations:               15028   AIC:                        -1.150e+04
Df Residuals:                   14969   BIC:                        -1.105e+04
Df Model:                          58                                         
Covariance Type:            nonrobust                                         
                                     coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------------
Intercept   

- R-squared: 0.576
- Positive interaction terms: 2006
- Not significant interaction terms: 2004, 2006, 2014, 2015, 2017
- Not significant covariance: drinking, all light condition factors, year_Q1, night

 # Y = Number of dead senior

In [19]:
mod8 = smf.ols(formula="S_DEAD ~ C(YEAR)+C(YEAR)*rounds_1", data = ct).fit()
print(mod8.summary())

                            OLS Regression Results                            
Dep. Variable:                 S_DEAD   R-squared:                       0.085
Model:                            OLS   Adj. R-squared:                  0.083
Method:                 Least Squares   F-statistic:                     42.32
Date:                Thu, 11 Jul 2019   Prob (F-statistic):          7.08e-260
Time:                        13:29:15   Log-Likelihood:                -4410.3
No. Observations:               15028   AIC:                             8889.
Df Residuals:                   14994   BIC:                             9148.
Df Model:                          33                                         
Covariance Type:            nonrobust                                         
                               coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------
Intercept               

- R-squared: 0.085
- Positive interaction terms: 2005
- Not significant interaction terms: 2005, 2006, 2008, 2017

In [20]:
mod9 = smf.ols(formula="S_DEAD ~ C(YEAR)+C(YEAR)*rounds_1+S_DRINKING+S_DRUG+WEATHER_GOOD+\
        WEATHER_RAIN+WEATHER_SLEET+WEATHER_SNOW+WEATHER_FOG+WEATHER_CLOUDY+LGT_COND_DAYLIGHT\
       +LGT_COND_DARK_NOT_LIGHTED+LGT_COND_DARK_LIGHTED+LGT_COND_DAWN+\
       LGT_COND_DUSK+LGT_COND_DARK_UNKNOWN_LIGHTING+WEEKDAY+\
       YEAR_Q1+YEAR_Q2+YEAR_Q3+MORNING+NOON+\
       AFTERNOON+NIGHT", data = ct).fit()
print(mod9.summary())

                            OLS Regression Results                            
Dep. Variable:                 S_DEAD   R-squared:                       0.445
Model:                            OLS   Adj. R-squared:                  0.443
Method:                 Least Squares   F-statistic:                     217.9
Date:                Thu, 11 Jul 2019   Prob (F-statistic):               0.00
Time:                        13:29:16   Log-Likelihood:                -661.25
No. Observations:               15028   AIC:                             1434.
Df Residuals:                   14972   BIC:                             1861.
Df Model:                          55                                         
Covariance Type:            nonrobust                                         
                                     coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------------
Intercept   

- R-squared: 0.445
- Positive interaction terms: 2005
- Not significant interaction terms: 2005, 2008, 2017
- Not significant covariance: sleet, snow, cloudy, weekday, night

In [21]:
mod10 = smf.ols(formula="S_DEAD ~ C(YEAR)+C(YEAR)*rounds_1+S_DRINKING+S_DRUG+WEATHER_GOOD+\
        WEATHER_RAIN+WEATHER_SLEET+WEATHER_SNOW+WEATHER_FOG+WEATHER_CLOUDY+LGT_COND_DAYLIGHT\
       +LGT_COND_DARK_NOT_LIGHTED+LGT_COND_DARK_LIGHTED+LGT_COND_DAWN+\
       LGT_COND_DUSK+LGT_COND_DARK_UNKNOWN_LIGHTING+WEEKDAY+\
       YEAR_Q1+YEAR_Q2+YEAR_Q3+MORNING+NOON+\
       AFTERNOON+NIGHT+rounds_2+rounds_3", data = ct).fit()
print(mod10.summary())

                            OLS Regression Results                            
Dep. Variable:                 S_DEAD   R-squared:                       0.450
Model:                            OLS   Adj. R-squared:                  0.447
Method:                 Least Squares   F-statistic:                     214.5
Date:                Thu, 11 Jul 2019   Prob (F-statistic):               0.00
Time:                        13:29:16   Log-Likelihood:                -593.19
No. Observations:               15028   AIC:                             1302.
Df Residuals:                   14970   BIC:                             1744.
Df Model:                          57                                         
Covariance Type:            nonrobust                                         
                                     coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------------
Intercept   

- R-squared: 0.450
- Positive interaction terms: 2005
- Not significant interaction terms: 2005, 2006, 2008, 2017
- Not significant covariance: sleet, snow, cloudy, weekday, night

In [22]:
mod11 = smf.ols(formula="S_DEAD ~ C(YEAR)+C(YEAR)*rounds_1+S_DRINKING+S_DRUG+WEATHER_GOOD+\
        WEATHER_RAIN+WEATHER_SLEET+WEATHER_SNOW+WEATHER_FOG+WEATHER_CLOUDY+LGT_COND_DAYLIGHT\
       +LGT_COND_DARK_NOT_LIGHTED+LGT_COND_DARK_LIGHTED+LGT_COND_DAWN+\
       LGT_COND_DUSK+LGT_COND_DARK_UNKNOWN_LIGHTING+WEEKDAY+\
       YEAR_Q1+YEAR_Q2+YEAR_Q3+MORNING+NOON+\
       AFTERNOON+NIGHT+rounds_2+rounds_3+S_MALE", data = ct).fit()
print(mod11.summary())

                            OLS Regression Results                            
Dep. Variable:                 S_DEAD   R-squared:                       0.607
Model:                            OLS   Adj. R-squared:                  0.606
Method:                 Least Squares   F-statistic:                     399.2
Date:                Thu, 11 Jul 2019   Prob (F-statistic):               0.00
Time:                        13:29:16   Log-Likelihood:                 1944.8
No. Observations:               15028   AIC:                            -3772.
Df Residuals:                   14969   BIC:                            -3322.
Df Model:                          58                                         
Covariance Type:            nonrobust                                         
                                     coef    std err          t      P>|t|      [95.0% Conf. Int.]
--------------------------------------------------------------------------------------------------
Intercept   

- R-squared: 0.607
- Positive interaction terms: none
- Not significant interaction terms: 2005
- Not significant covariance: drinking, dark_not_lighted, dark_lighted, dusk, dark and unknown lighting, morning, afternoon, night